tests/test_views.py

```diff
@@ -0,0 +1,77 @@
+"""Tests for the views functions."""
+from datetime import datetime
+from unittest.mock import patch
+
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+from django.test import TestCase, Client
+from django.utils import timezone
+
+
+from blog.models import Post
+
+
+class ViewsTest(TestCase):
+    """Initialize the test client and user."""
+    def setUp(self):
+        self.client = Client()
+        self.user = get_user_model().objects.create()
+
+    def test_post_list_rendering(self):
+        """Test rendering of posts with different dates mocked."""
+        tm_zone = timezone.get_current_timezone()
+        post = Post.objects.create(
+            author=self.user, title='current title', text='current text',
+            created_date=datetime(day=6, month=2, year=2024, tzinfo=tm_zone),
+            published_date=datetime(day=6, month=2, year=2024, tzinfo=tm_zone),
+        )
+        past_post = Post.objects.create(
+            author=self.user, title='past title', text='past text',
+            created_date=datetime(day=5, month=2, year=2024, tzinfo=tm_zone),
+            published_date=datetime(day=5, month=2, year=2024, tzinfo=tm_zone),
+        )
+        future_post = Post.objects.create(
+            author=self.user, title='future title', text='future text',
+            created_date=datetime(day=7, month=2, year=2025, tzinfo=tm_zone),
+            published_date=datetime(day=7, month=2, year=2025, tzinfo=tm_zone),
+        )
+
+        with patch(
+            'django.utils.timezone.now',
+            lambda: datetime(day=5, month=2, year=2024, tzinfo=tm_zone),
+        ):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(list(response.context['posts']), [past_post])
+            self.assertNotContains(response, post)
+            self.assertNotContains(response, future_post)
```

Comment:
Use `self.subTest()` for every distinct check.

Reply:
Added

Reply:
Marking this thread as resolved.